x-pack/legacy/plugins/alerting/server/alerts_client.ts

```diff
@@ -23,26 +23,32 @@ import {
 } from './types';
 import { TaskManagerStartContract } from './shim';
 import { validateAlertTypeParams } from './lib';
-import { CreateAPIKeyResult as SecurityPluginCreateAPIKeyResult } from '../../../../plugins/security/server';
+import {
+  InvalidateAPIKeyParams,
+  CreateAPIKeyResult as SecurityPluginCreateAPIKeyResult,
+  InvalidateAPIKeyResult as SecurityPluginInvalidateAPIKeyResult,
+} from '../../../../plugins/security/server';
+import { PluginStartContract as EncryptedSavedObjectsStartContract } from '../../../../plugins/encrypted_saved_objects/server';
 
-interface FailedCreateAPIKeyResult {
-  created: false;
-}
-interface SuccessCreateAPIKeyResult {
-  created: true;
-  result: SecurityPluginCreateAPIKeyResult;
-}
-export type CreateAPIKeyResult = FailedCreateAPIKeyResult | SuccessCreateAPIKeyResult;
 type NormalizedAlertAction = Omit<AlertAction, 'actionTypeId'>;
+export type CreateAPIKeyResult =
+  | { apiKeysEnabled: false }
+  | { apiKeysEnabled: true; result: SecurityPluginCreateAPIKeyResult };
+export type InvalidateAPIKeyResult =
+  | { apiKeysEnabled: false }
+  | { apiKeysEnabled: true; result: SecurityPluginInvalidateAPIKeyResult };
 
 interface ConstructorOptions {
   logger: Logger;
   taskManager: TaskManagerStartContract;
   savedObjectsClient: SavedObjectsClientContract;
   alertTypeRegistry: AlertTypeRegistry;
+  encryptedSavedObjectsPlugin: EncryptedSavedObjectsStartContract;
   spaceId?: string;
+  namespace?: string;
```

Comment:
What's the difference between `spaceId` and `namespace`?  Looks like `spaceId` is only ever used in `scheduleAlert()`.  I see some a conversion function used elsewhere in files in the PR - `spaceIdToNamespace()`, wondering if we should just use `spaceIdToNamespace(this.spaceId)` instead of adding the `namespace` parameter?

Reply:
I'm not sure myself of the difference but I have been told by the security team that there is a difference. Also that we should use `spaceId` as our source of truth to convert into `namespace` and `basePath`.

We can use the function `spaceIdToNamespace` to get the value but I would have to pass the `spaceIdToNamespace` function as an argument to alerts client instead. So I figured I might as well pass the result of the function. I'm indifferent on the approach, thoughts?

Reply:
eh, I worry about two highly related values being stored like that, in case one gets "out of sync" with the other, but I don't think that's really an issue since you can't really change the space of a SO anyway.  And having to send more functions in doesn't sound all that great either.  Seems like leaving it the way it is, is fine.

Reply:
I'm hoping eventually we can just pass around the setup and start objects instead of sub-functions / properties. Though, not sure if that's good or bad but would reduce the number of things being passed around drastically.